packages/select/src/__tests__/SingleSelect.spec.tsx

```diff
@@ -0,0 +1,98 @@
+import { Item } from '@react-stately/collections';
+import { it, expect, describe } from 'vitest';
+
+import { SingleSelect, SingleSelectTrigger } from '..';
+import { render, screen, userEvent } from '../../../../test/utils';
+
+import { FRUIT } from './constants';
+import { CustomSingleSelectTrigger } from './examples';
+
+describe('Select', () => {
+  describe('with default single select trigger', () => {
+    it('renders', () => {
+      render(
+        <SingleSelect label="Fruit" items={FRUIT}>
+          {(item) => <Item>{item.name}</Item>}
+        </SingleSelect>
+      );
+      expect(screen.getByTestId('select')).toBeVisible();
+    });
+
+    it('opens menu, selects item, and updates selected value', async () => {
+      const user = userEvent.setup();
+      render(
+        <SingleSelect label="Fruit" items={FRUIT}>
+          {(item) => <Item>{item.name}</Item>}
+        </SingleSelect>
+      );
+      await user.click(screen.getByTestId('select-trigger'));
+      await user.click(screen.getAllByRole('option')[0]);
+      expect(screen.getByTestId('select-trigger')).toHaveTextContent(FRUIT[0].name);
+    });
+
+    it('renders no header by default', async () => {
+      const user = userEvent.setup();
+
+      render(
+        <SingleSelect label="Fruit" items={FRUIT}>
+          {(item) => <Item>{item.name}</Item>}
+        </SingleSelect>
+      );
+
+      await user.click(screen.getByTestId('select-trigger'));
+
+      expect(screen.queryByTestId('menu-header')).toBeNull();
+    });
+  });
+
+  describe('with default single select trigger passed custom selected value', () => {
+    it('renders with custom value', async () => {
+      const user = userEvent.setup();
+      const DUMMY_TEXT = 'Selected value dummy text';
+
+      render(
+        <SingleSelect
+          label="Fruit"
+          trigger={(props) => (
+            <SingleSelectTrigger {...props}>{() => DUMMY_TEXT}</SingleSelectTrigger>
+          )}
+          items={FRUIT}
+        >
+          {(item) => <Item>{item.name}</Item>}
+        </SingleSelect>
+      );
+
+      expect(screen.getByTestId('select')).toBeVisible();
+
+      await user.click(screen.getByTestId('select-trigger'));
```

Comment:
we probably want to consider duplicating these same test scenarios, but firing with keyboard events instead